### PR TITLE
add rest_parameters as definition in javascript

### PR DESCRIPTION
**Pull request by steelsojka**

### CODE REVIEW

**Comment by kyazdani42** [comment]
i'm not too sure but aren't rest parameters `@parameters` ?

**Comment by vigoux** [comment]
Welcome back @kyazdani42 !

**Comment by steelsojka** [comment]
@kyazdani42 Yes, but it also a definition like other parameters.

**Comment by steelsojka** [comment]
I should also add them as a parameter for highlighting though...

**Comment by kyazdani42** [comment]
go for it

File: queries/javascript/locals.scm [hunk 1]
```diff
@@ -25,6 +25,10 @@
   (array_pattern
     (identifier) @definition))
 
+(formal_parameters
+  (rest_parameter
+    (identifier) @definition))
```
Comment (theHamsta) [inline queries/javascript/locals.scm:30]: @definition.var ? Also for other variable definitions.

File: queries/javascript/highlights.scm [hunk 2]
```diff
@@ -74,6 +74,10 @@
 
 (formal_parameters (identifier) @variable.parameter)
 
+(formal_parameters
+  (rest_parameter
+    (identifier) @variable.parameter))
```
Comment (theHamsta) [inline queries/javascript/highlights.scm:79]: @parameter

**Pull request merged**